lua/cfc_time/server/time/sv_time_tracking.lua

```diff
@@ -0,0 +1,81 @@
+CFCTime.ctime = {}
+local ctime = CFCTime.ctime
+
+local function now()
+    return os.time()
+end
+
+-- <steamID64> = { joined = <timestamp>, departed = <timestamp> | nil, initialTime = <float> }
+ctime.pendingUpdates = {}
+ctime.updateTimerName = "CFC_Time_UpdateTimer"
+ctime.lastUpdate = now()
+
+function ctime:updateTimes()
+    local batch = {}
+    local now = now()
+
+    for steamId, data in pairs( self.pendingUpdates ) do
+        local isValid = true
+        local joined = data.joined
+        local departed = data.departed
+        local initialTime = data.initialTime
+
+        if departed < self.lastUpdate then
+            self.pendingUpdates[steamId] = nil
+            isValid = false
+        end
+
+        if isValid then
+            local sessionTime = now - joined
+            local newTime = initialTime + sessionTime
+
+            batch[steamId] = newTime
+        end
+    end
+
+    CFCTime.SQL:UpdateTimeBatch( batch )
+    ctime.lastUpdate = now
+end
+
+function ctime:startTimer()
+    timer.Create(
+        self.updateTimerName,
+        self.Config.updateInterval,
+        0,
+        function() ctime:updateTimes() end
```

Comment:
Since this is already an anonymous function, it may be a good call to use pcall here, as timers permanently die when an error occurs.

*Not saying it would but...*